clients/js/src/index.ts

```diff
@@ -440,60 +658,172 @@ export class Collection {
+   *
+   * @example
+   * ```typescript
+   * const client = new ChromaClient({
+   *   path: "http://localhost:8000"
+   * });
+   * ```
+   */
+  constructor({ path }: { path?: string } = {}) {
+    if (path === undefined) path = "http://localhost:8000";
     const apiConfig: Configuration = new Configuration({
-      basePath,
+      basePath: path,
     });
     this.api = new DefaultApi(apiConfig);
   }
 
-  public async reset() {
+  /**
+   * Resets the state of the object by making an API call to the reset endpoint.
+   *
+   * @returns {Promise<Api.Reset200Response>} A promise that resolves when the reset operation is complete.
+   * @throws {Error} If there is an issue resetting the state.
+   *
+   * @example
+   * ```typescript
+   * await client.reset();
+   * ```
+   */
+  public async reset(): Promise<Api.Reset200Response> {
     return await this.api.reset();
   }
 
-  public async version() {
+  /**
+   * Returns the version of the Chroma API.
+   * @returns {Promise<string>} A promise that resolves to the version of the Chroma API.
+   *
+   * @example
+   * ```typescript
+   * const version = await client.version();
+   * ```
+   */
+  public async version(): Promise<string> {
     const response = await this.api.version();
     return await handleSuccess(response);
   }
 
-  public async heartbeat() {
+  /**
+   * Returns a heartbeat from the Chroma API.
+   * @returns {Promise<number>} A promise that resolves to the heartbeat from the Chroma API.
+   *
+   * @example
+   * ```typescript
+   * const heartbeat = await client.heartbeat();
+   * ```
+   */
+  public async heartbeat(): Promise<number> {
     const response = await this.api.heartbeat();
     let ret = await handleSuccess(response);
     return ret["nanosecond heartbeat"]
   }
 
-  public async persist() {
+  /**
+   * @ignore
+   */
+  public async persist(): Promise<never> {
     throw new Error("Not implemented in JS client");
   }
 
-  public async createCollection(
+  /**
+   * Creates a new collection with the specified properties.
+   *
+   * @param {Object} params - The parameters for creating a new collection.
+   * @param {string} params.name - The name of the collection.
+   * @param {Metadata} [params.metadata] - Optional metadata associated with the collection.
+   * @param {CallableFunction} [params.embeddingFunction] - Optional custom embedding function for the collection.
+   *
+   * @returns {Promise<Collection>} A promise that resolves to the created collection.
+   * @throws {Error} If there is an issue creating the collection.
+   *
+   * @example
+   * ```typescript
+   * const collection = await client.createCollection({
+   *   name: "my_collection",
+   *   metadata: {
+   *     "description": "My first collection"
+   *   }
+   * });
+   * ```
+   */
+  public async createCollection({
+    name,
+    metadata,
+    embeddingFunction
+  }: {
     name: string,
-    metadata?: object,
+    metadata?: Metadata,
```

Comment:
this metadata has the same type -> break out a new a new type `CollectionMetadata`

Reply:
persist through all locations

Reply:
Done